omniscidb/Tests/ResultSetArrowConversion.cpp

```diff
@@ -385,6 +385,111 @@ TEST(ArrowTable, EmptySelection) {
   ASSERT_EQ(table->num_rows(), (int64_t)0);
 }
 
+TEST(ArrowTable, LimitedSelection) {
+  auto res = runSqlQuery("select i, count(bi) from test group by i limit 4 offset 1;",
+                         ExecutorDeviceType::CPU,
+                         true);
+  ASSERT_EQ(res.getRows()->rowCount(), (int64_t)1);
+
+  auto table = getArrowTable(res);
+  ASSERT_NE(table, nullptr);
+  ASSERT_EQ(table->num_columns(), 2);
+  ASSERT_EQ(table->num_rows(), (int64_t)1);
+
+  std::array<int32_t, 1> res_arr_0{1};
+  std::array<int64_t, 1> res_arr_1{3};
+  compare_columns(res_arr_0, table->column(0));
+  compare_columns(res_arr_1, table->column(1));
+}
+
+TEST(ArrowTable, OrderedLimitedSelection) {
+  auto res = runSqlQuery("select d, bi from test order by bi desc limit 4 offset 2;",
+                         ExecutorDeviceType::CPU,
+                         true);
+  ASSERT_EQ(res.getRows()->rowCount(), (int64_t)4);
+
+  auto table = getArrowTable(res);
+  ASSERT_NE(table, nullptr);
+  ASSERT_EQ(table->num_columns(), 2);
+  ASSERT_EQ(table->num_rows(), (int64_t)4);
+
+  std::array<double, 6> sorted_arr_0 = table6x4_col_d;
+  std::sort(sorted_arr_0.begin(), sorted_arr_0.end(), std::greater{});
+
+  std::array<double, 4> res_arr_0;
+  std::copy(sorted_arr_0.begin() + 2, sorted_arr_0.begin() + 6, res_arr_0.begin());
+  compare_columns(res_arr_0, table->column(0));
+
+  std::array<int64_t, 6> sorted_arr_1 = table6x4_col_bi;
+  std::sort(sorted_arr_1.begin(), sorted_arr_1.end(), std::greater{});
+
+  std::array<int64_t, 4> res_arr_1;
+  std::copy(sorted_arr_1.begin() + 2, sorted_arr_1.begin() + 6, res_arr_1.begin());
+  compare_columns(res_arr_1, table->column(1));
+}
+
+TEST(ArrowTable, ColumnarLimitedJoin) {
+  bool prev_enable_columnar_output = config().rs.enable_columnar_output;
+  bool prev_enable_lazy_fetch = config().rs.enable_lazy_fetch;
+
+  ScopeGuard reset = [prev_enable_columnar_output, prev_enable_lazy_fetch] {
+    config().rs.enable_columnar_output = prev_enable_columnar_output;
+    config().rs.enable_lazy_fetch = prev_enable_lazy_fetch;
+  };
+
+  config().rs.enable_columnar_output = true;
+  config().rs.enable_lazy_fetch = false;
+  auto res = runSqlQuery(
+      "SELECT * FROM test INNER JOIN join_table ON test.i=join_table.i offset 4 limit 2;",
```

Comment:
For all non-groupby tests use `test_chunked` or run tests twice, once on `test` and another one on `test_chunked`. The chunked table holds the same data but is split into different fragments. This will produce ResultSets with multiple storages and improve your testing coverage.